errors/errors.go

```diff
@@ -184,6 +184,9 @@ var (
 	ErrInvalidComponentProviders              = errors.New("invalid component providers section")
 	ErrInvalidComponentHooks                  = errors.New("invalid component hooks section")
 	ErrInvalidComponentAuth                   = errors.New("invalid component auth section")
+	ErrMergeComponentAuth                     = errors.New("failed to merge component auth config")
```

Comment:
It seems like we should have a general merge error, and then we wrap it with the context that this is an auth config.